pkg/resourcecollector/persistentvolumeclaim.go

```diff
@@ -83,9 +83,11 @@ func (r *ResourceCollector) preparePVCResourceForApply(
 		if vol.PersistentVolumeClaim == pvc.Name {
 			for _, node := range nodes.Items {
 				nodeZone := node.Labels[v1.LabelTopologyZone]
-				if nodeZone == vol.Zones[0] {
-					pvc.Annotations[pvutil.AnnSelectedNode] = node.Name
-					break
+				if len(vol.Zones) != 0 {
+					if nodeZone == vol.Zones[0] {
+						pvc.Annotations[pvutil.AnnSelectedNode] = node.Name
```

Comment:
nit: create the pvc.Annotations map if it is nil
you can have a pvc with no annotations

Reply:
Fixed